allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java

```diff
@@ -99,45 +96,45 @@ public Allure2ResultsReader(AttachmentsStorage storage) {
                         dest.setTestStage(testStage);
                     }
 
-                    dest.getBeforeStages().addAll(getParentBefores(result.getParentIds(), groups));
-                    dest.getBeforeStages().addAll(convert(result.getBefores(), this::convert));
-                    dest.getAfterStages().addAll(convert(result.getAfters(), this::convert));
-                    dest.getAfterStages().addAll(getParentAfters(result.getParentIds(), groups));
+                    List<TestResultContainer> parents = findAllParents(groups, result.getId(), new HashSet<>());
```

Comment:
better to move everything from `.map(...)` into a separate method, as it contradicts streams' conciseness logic